### PR TITLE
Refactor test suite and fix XPath bug in formatter specs

### PR DESCRIPTION
**Pull request by eitoball**
## Summary

- Fix a critical XPath bug: `'/error'` matched nothing under `<file>`, so severity mapping was never actually tested
- Replace complex `Cop::Cop` setup with direct `RuboCop::Cop::Offense` construction via a `build_offense` helper
- Add tests for offense attributes (line, column, message, source) and empty offense lists
- Modernize `spec_helper.rb` (`__dir__`, `frozen_string_literal`) and exclude specs from RuboCop metrics cops

## Test plan

- [x] All specs pass (`bundle exec rspec spec`)
- [x] RuboCop passes on `spec/` (`bundle exec rubocop spec`)

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: spec/rubocop/formatter/checkstyle_formatter_spec.rb [hunk 1]
```diff
@@ -1,66 +1,98 @@
...
+                       end
+
+            expect(severity).to eq(expected)
           end
         end
+
+        it 'uses a relative path by default' do
+          file_element = REXML::XPath.first(doc, '//checkstyle/file')
+          expect(file_element.attribute('name').value).to eq('sample.rb')
+        end
```
[the remark this answers is not in the document]
Comment (eitoball) [reply spec/rubocop/formatter/checkstyle_formatter_spec.rb:66]: Good catch — fixed. The "uses a relative path by default" example now temporarily removes RUBOCOP_CHECKSTYLE_FORMATTER_ABSOLUTE_PATH before asserting and restores it afterward, so the example reliably exercises the unset/default state regardless of the caller's environment.

File: spec/rubocop/formatter/checkstyle_formatter_spec.rb [hunk 2]
```diff
@@ -1,66 +1,98 @@
...
+      end
+
+      describe 'offense attributes' do
+        let(:offense) { build_offense(:warning, line: 3, column: 4, message: 'unused variable') }
+        let(:error) { REXML::XPath.first(format_file([offense]), '//checkstyle/file/error') }
+
+        it 'includes line, column, message, and source' do
+          expect(error.attribute('line').value).to eq('3')
+          expect(error.attribute('column').value).to eq('4')
```
[the remark this answers is not in the document]
Comment (eitoball) [reply spec/rubocop/formatter/checkstyle_formatter_spec.rb]: Good catch — fixed. The formatter now emits offense.real_column (1-based) instead of offense.column (0-based), and the spec expectation was updated accordingly (column: 4 now expects real_column 5).
